indra/llcommon/llsdserialize.cpp

```diff
@@ -2220,54 +2228,67 @@ std::string zip_llsd(LLSD& data)
 // not very efficient -- creats a copy of decompressed LLSD block in memory
 // and deserializes from that copy using LLSDSerialize
 LLUZipHelper::EZipRresult LLUZipHelper::unzip_llsd(LLSD& data, std::istream& is, S32 size)
+{
+	std::unique_ptr<U8[]> in = std::unique_ptr<U8[]>(new(std::nothrow) U8[size]);
+	if (!in)
+	{
+		return ZR_MEM_ERROR;
+	}
+	is.read((char*) in.get(), size); 
+
+	return unzip_llsd(data, in.get(), size);
+}
+
+LLUZipHelper::EZipRresult LLUZipHelper::unzip_llsd(LLSD& data, const U8* in, S32 size)
 {
 	U8* result = NULL;
-	llssize cur_size = 0;
+	U32 cur_size = 0;
 	z_stream strm;
 		
-	const U32 CHUNK = 65536;
+	constexpr U32 CHUNK = 1024 * 512;
 
-	U8 *in = new(std::nothrow) U8[size];
-	if (!in)
+	static thread_local std::unique_ptr<U8[]> out;
```

Comment:
I understand why this must be `thread_local` if it's `static`; but I don't understand why it must be `static`?

Also, this might be a better use case for `std::vector<U8>`.